File: functions/test/veryfiAnyDocConstants.test.js
"use strict";

const test = require("node:test");
const assert = require("node:assert/strict");
const {
  VERYFI_ANY_DOCS_BASE_URL,
  VERYFI_ANY_DOCS_PATH,
  VERYFI_ANY_DOCS_URL,
  BLUEPRINT_HEALTH_INSURANCE_EOB,
  DOCUMENT_TYPE_EOB,
  CATEGORY_INSURANCE,
  CATEGORIES_INSURANCE
} = require("../lib/veryfiAnyDocConstants");

test("AnyDocs URL mirrors Android VeryfiAnyDocConstants", () => {
  assert.equal(VERYFI_ANY_DOCS_BASE_URL, "https://api.veryfi.com/api/v8/");
  assert.equal(VERYFI_ANY_DOCS_PATH, "partner/documents/");
  assert.equal(VERYFI_ANY_DOCS_URL, "https://api.veryfi.com/api/v8/partner/documents/");
});

test("health_insurance_eob blueprint is configured for EOB AnyDocs extraction", () => {
  assert.equal(BLUEPRINT_HEALTH_INSURANCE_EOB, "health_insurance_eob");
});

test("EOB document_type and insurance categories mirror Android AnyDocs contract", () => {
  assert.equal(DOCUMENT_TYPE_EOB, "eob");
  assert.equal(CATEGORY_INSURANCE, "insurance");
  assert.deepEqual(CATEGORIES_INSURANCE, ["insurance"]);
});

File: app/src/test/java/app/eob/me/VeryfiAnyDocConstantsTest.kt
package app.eob.me

import app.eob.me.network.VeryfiAnyDocConstants
import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

/**
 * Guards AnyDocs endpoint URL and blueprint strings against drift that would cause HTTP 404s.
 */
class VeryfiAnyDocConstantsTest {
    @Test
    fun documentsUrlComposesToPartnerDocumentsEndpoint() {
        val resolved = VeryfiAnyDocConstants.BASE_URL + VeryfiAnyDocConstants.ANY_DOCUMENTS_PATH
        assertEquals("https://api.veryfi.com/api/v8/partner/documents/", resolved)
        assertTrue(resolved.startsWith("https://api.veryfi.com/api/v8/"))
        assertTrue(resolved.endsWith("partner/documents/"))
    }

    @Test
    fun healthInsuranceEobBlueprintMatchesAnyDocsContract() {
        assertEquals("health_insurance_eob", VeryfiAnyDocConstants.BLUEPRINT_HEALTH_INSURANCE_EOB)
        assertFalse(VeryfiAnyDocConstants.BLUEPRINT_HEALTH_INSURANCE_EOB.isBlank())
    }

    @Test
    fun hybridStreamCallableMatchesCloudFunctionExport() {
        assertEquals("extractVeryfiHybridStream", VeryfiAnyDocConstants.EXTRACT_VERYFI_HYBRID_STREAM)
    }

    @Test
    fun eobDocumentTypeAndInsuranceCategoriesMatchAnyDocsContract() {
        assertEquals("eob", VeryfiAnyDocConstants.DOCUMENT_TYPE_EOB)
        assertEquals(listOf("insurance"), VeryfiAnyDocConstants.CATEGORIES_INSURANCE)
    }
}
